### PR TITLE
Unify intervals notation

### PR DESCRIPTION
**Pull request by NeilZaim**
Following #1346 we have decided to unify the `ìnt`, `period`, `frequency` and `intervals` notations into `intervals`, which is done in this PR. This leads to the following changes in the input file:

- `warpx.load_balance_int` => `warpx.load_balance_intervals`
- `warpx.override_sync_int` => `warpx.override_sync_intervals`
- `warpx.sort_int` => `warpx.sort_intervals`
- `<diag_name>.period` => `<diag_name>.intervals`
- `<reduced_diag_name>.frequency` => `<reduced_diag_name>.intervals`

I haven't found any other input parameter that should be changed, but maybe I missed some.

Also note that I'm not really familiar with PICMI so it might be worth double-checking these parts in particular to make sure that I haven't done anything wrong with the python layer.

### CODE REVIEW

**Comment by RemiLehe** [comment]
@NeilZaim Thanks a lot for this PR! Could you double-check why the automated tests are failing for this PR?

**Comment by NeilZaim** [comment]
@RemiLehe I've had a look today and I think I understand what's wrong. I'll push a new commit tomorrow.

**Comment by NeilZaim** [comment]
Looks like it passes the test now. It's probably still worth triple-checking the PICMI part, it's the first time I modify that.

**Comment by ax3l** [comment]
@NeilZaim can you please rebase once more?

@dpgrote do you like to review the PICMI part?

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -777,7 +778,7 @@ def initialize_inputs(self):
         self.diagnostic.format = self.format
         self.diagnostic.openpmd_backend = self.openpmd_backend
         self.diagnostic.dump_rz_modes = self.dump_rz_modes
-        self.diagnostic.period = self.period
+        self.diagnostic.intervals = self.intervals
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py]: This should continue to use the PICMI standard for input, so this line should be
`self.diagnostics.intervals = self.period`

File: Python/pywarpx/picmi.py [hunk 2]
```diff
@@ -759,6 +759,7 @@ def init(self, kw):
         self.openpmd_backend = kw.pop('warpx_openpmd_backend', None)
         self.file_prefix = kw.pop('warpx_file_prefix', None)
         self.dump_rz_modes = kw.pop('warpx_dump_rz_modes', None)
+        self.intervals = kw.pop('warpx_intervals', None)
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py]: With the fix below, this line is not needed.

File: Python/pywarpx/picmi.py [hunk 3]
```diff
@@ -882,7 +884,7 @@ def initialize_inputs(self):
         self.diagnostic.diag_type = 'Full'
         self.diagnostic.format = self.format
         self.diagnostic.openpmd_backend = self.openpmd_backend
-        self.diagnostic.period = self.period
+        self.diagnostic.intervals = self.intervals
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py]: This should also be
`self.diagnostics.intervals = self.period`

File: Python/pywarpx/picmi.py [hunk 4]
```diff
@@ -854,6 +855,7 @@ def init(self, kw):
         self.random_fraction = kw.pop('warpx_random_fraction', None)
         self.uniform_stride = kw.pop('warpx_uniform_stride', None)
         self.plot_filter_function = kw.pop('warpx_plot_filter_function', None)
+        self.intervals = kw.pop('warpx_intervals', None)
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py]: This line is not needed.

**Review by dpgrote — COMMENTED** [review]
Thanks for the clean up. I have some comments in the picmi.py file.

**Comment by MaxThevenet** [comment]
Any update on this PR? It breaks backward-compatibility, but it gives a nice message and gives more consistency, so it looks like a good change.

**Comment by NeilZaim** [comment]
@dpgrote Thanks for the review! Your comments should be taken into account now.

@MaxThevenet Sorry, I was a bit busy last week but this PR should be ready now.

**Review by MaxThevenet — APPROVED** [review]

**Comment by MaxThevenet** [comment]
Let us coordinate to merge related non-backward-compatible changes like PR https://github.com/ECP-WarpX/WarpX/pull/1530 together.

**Comment by ax3l** [comment]
agreed, shall we merge #1485 first?

**Review by ax3l — APPROVED** [review]
LGTM, thank you for this :tada:

**Comment by ax3l** [comment]
@NeilZaim just a small conflict in the Langmuior 3D input left - it's removed in the latest `development`, so no need to change it :)

**Comment by NeilZaim** [comment]
@ax3l This should be good now

**Comment by ax3l** [comment]
@NeilZaim, we are close, sorry a new little conflict ;-)

**Comment by NeilZaim** [comment]
@ax3l Conflict fixed :)

**Review by ax3l — APPROVED** [review]

**Pull request merged**